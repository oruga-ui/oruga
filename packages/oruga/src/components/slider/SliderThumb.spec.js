import { mount } from '@vue/test-utils'
import OSlider from '@components/slider/Slider'

let wrapper
let wrapperParent

describe('OSliderThumb', () => {
    beforeEach(() => {
        wrapperParent = mount(OSlider, {
            propsData: {
                step: 5,
            }
        })
        wrapper = wrapperParent.find({ ref: 'button1' })
    })

    it('is called', () => {
        expect(wrapper.exists()).toBeTruthy()
    })

    it('render correctly', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })

    it('inherits parent properties', () => {
        expect(wrapper.vm.step).toBe(5)
    })

    it('manage focus', () => {
        wrapper.vm.onFocus()
        expect(wrapper.vm.isFocused).toBeTruthy()

        wrapper.vm.onBlur()
        expect(wrapper.vm.isFocused).toBeFalsy()
    })
})
