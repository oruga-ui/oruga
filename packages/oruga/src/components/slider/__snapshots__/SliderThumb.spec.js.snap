// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OSliderThumb render correctly 1`] = `
<div class="o-slide__thumb-wrapper" style="left: 0%;">
    <div class="o-tip">
        <div class="o-tip__content o-tip__content--top" style="display: none;"><span class="o-tip__arrow o-tip__arrow--top"></span> 0</div>
        <div class="o-tip__trigger">
            <div tabindex="0" role="slider" aria-valuenow="0" aria-valuemin="0" aria-valuemax="100" aria-orientation="horizontal" class="o-slide__thumb">
                <!---->
            </div>
        </div>
    </div>
</div>
`;
